springcloud/src/main/java/org/axonframework/extensions/springcloud/commandhandling/SpringHttpCommandBusConnector.java

```diff
@@ -226,24 +232,27 @@ public Optional<CommandBus> localSegment() {
     @PostMapping("/command")
     public <C, R> CompletableFuture<?> receiveCommand(@RequestBody SpringHttpDispatchMessage<C> dispatchMessage) {
         CommandMessage<C> commandMessage = dispatchMessage.getCommandMessage(serializer);
-        if (dispatchMessage.isExpectReply()) {
-            try {
-                SpringHttpReplyFutureCallback<C, R> replyFutureCallback = new SpringHttpReplyFutureCallback<>();
-                localCommandBus.dispatch(commandMessage, replyFutureCallback);
-                return replyFutureCallback;
-            } catch (Exception e) {
-                logger.error("Could not dispatch command", e);
-                return CompletableFuture.completedFuture(createReply(commandMessage, asCommandResultMessage(e)));
-            }
-        } else {
-            try {
-                localCommandBus.dispatch(commandMessage);
-                return CompletableFuture.completedFuture("");
-            } catch (Exception e) {
-                logger.error("Could not dispatch command", e);
-                return CompletableFuture.completedFuture(createReply(commandMessage, asCommandResultMessage(e)));
+        Span span = spanFactory.createChildHandlerSpan(() -> "SpringHttpCommandBusConnector.handle", commandMessage);
+        return span.runSupplier(() -> {
+            if (dispatchMessage.isExpectReply()) {
+                try {
+                    SpringHttpReplyFutureCallback<C, R> replyFutureCallback = new SpringHttpReplyFutureCallback<>();
+                    localCommandBus.dispatch(commandMessage, replyFutureCallback);
+                    return replyFutureCallback;
+                } catch (Exception e) {
+                    logger.error("Could not dispatch command", e);
```

Comment:
Catching the exception here means it will no longer be recorded on the span, Which is likely not the wanted behavior.

Reply:
Ah yes. span.recordException can be called here

Reply:
Good catch! Added `Span#recordException` to this try-catch clause and the one starting on line 247.